src/modm/platform/can/atsam-fdcan/can.hpp

```diff
@@ -0,0 +1,433 @@
+/*
+ * Copyright (c) 2019, 2023, Raphael Lehmann
+ * Copyright (c) 2021, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include <optional>
+
+#include <modm/architecture/interface/can.hpp>
+#include <modm/architecture/interface/can_filter.hpp>
+#include <modm/platform/gpio/connector.hpp>
+
+#include "can_bit_timings.hpp"
+
+#include "message_ram.hpp"
+
+namespace {
+using ErrorCallback = void (*)();
+
+extern ErrorCallback mcan0ErrorCallback;
+extern ErrorCallback mcan1ErrorCallback;
+}
```

Comment:
`mcan0ErrorCallback` is defined inside namespace `modm::platform` in can_interrupt.cpp. The `extern` declarations here are in an anonymous namespaces and can't refer to the definitions in can_interrupt.cpp. If `setErrorInterruptCallback()` was actually used and instantiated the result would be a linker error. 

An anonymous namespaces gives symbols defined in it internal linkage, like `static`. Therefore, `extern` declarations in an anonymous namespace are nonsensical.